File: Ordenamiento c++/bubble/bubble.cpp
#include <iostream>


void swap(int *xp, int *yp)  
{  
    int temp = *xp;  
    *xp = *yp;  
    *yp = temp;  
}  


void burbuja(int *A,int n)
{
	int i, j;  
    for (i = 0; i < n-1; i++)      
    	for (j = 0; j < n-i-1; j++)  
        	if (A[j] > A[j+1])  
            	swap(&A[j], &A[j+1]);
}
int main()
{
	int tam = 100 ;
	int *a = new int[tam];
	for(int k=0;k<tam;k++) 
	{
        	std::cin>>a[k];            
	} 
	burbuja(a, tam );
	for(int i=0;i<tam;i++)
        std::cout<<a[i]<<std::endl;
	delete []a;
	return 0;
}

File: Sorts/Bubble/bubble.java
class Bubble{ 
	void burbuja(int A[],int n)
	{
		for(int i=0;i<n;i++)
		{
			for(int j=0;j<n-i-1;j++)
			{
				if(A[j]>A[j+1])
				{
					int temp=A[j];
					A[j]=A[j+1];
					A[j+1]=temp;
				}
			}
		}
	}
	void mostrar(int A[],int n)
	{
		for(int i=0;i<n;i++)
		{
			System.out.println(A[i]+" ");
		}
		System.out.println();
	}
    public static void main(String args[]) 
    {     
    	Bubble prueba=new Bubble();
    	int A[]={10,9,8,7,6,5,4,3,2,1};
    	prueba.burbuja(A,10);
    	prueba.mostrar(A,10);
        
    }
}